src/commands/app/pack.js

```diff
@@ -0,0 +1,266 @@
+   * @param {boolean} pathInZip internal path in zip
+   * @returns {Promise} returns with a blank promise when done
+   */
+  zipHelper (filePath, out, pathInZip = false) {
+    aioLogger.debug(`Creating zip of file/folder '${filePath}'`)
+    const stream = fs.createWriteStream(out)
+    const archive = archiver('zip', { zlib: { level: 9 } })
+
+    return new Promise((resolve, reject) => {
+      stream.on('close', () => resolve())
+      archive.pipe(stream)
+      archive.on('error', err => reject(err))
+
+      let stats
+      try {
+        stats = fs.lstatSync(filePath) // throws if enoent
+      } catch (e) {
+        archive.destroy()
+        reject(e)
+      }
+
+      if (stats.isDirectory()) {
+        archive.directory(filePath, pathInZip)
+      } else { //  if (stats.isFile()) {
+        archive.file(filePath, { name: pathInZip || path.basename(filePath) })
+      }
+      archive.finalize()
+    })
+  }
+
+  /**
+   * Gets a list of files that are to be packed.
+   *
+   * This runs `npm pack` to get the list.
+   *
+   * @param {string} workingDirectory the working directory to run `npm pack` in
+   * @returns {Array<string>} a list of files that are to be packed
+   */
+  async filesToPack (workingDirectory = process.cwd()) {
+    const { stdout } = await execa('npm', ['pack', '--dry-run', '--json'], { cwd: workingDirectory })
+
+    const { files } = JSON.parse(stdout)[0]
+    return files.map(file => file.path)
+  }
+
+  /**
+   * An annotation called code-download will be added to all actions in app.config.yaml
+   * (and linked yaml configs for example in extensions). This value will be set to false.
+   * The annotation will by default be true if not set.
+   *
+   * @param {object} appConfig the app's configuration file
+   */
+  async addCodeDownloadAnnotation (appConfig) {
+    // get the configFiles that have runtime manifests
+    const configFiles = []
+    for (const [, value] of Object.entries(appConfig.includeIndex)) {
+      const { key } = value
+      if (key === 'runtimeManifest' || key === 'application.runtimeManifest') {
+        configFiles.push(value)
+      }
+    }
+
+    // for each configFile, we modify each action to have the "code-download: false" annotation
+    for (const configFile of configFiles) {
+      const configFilePath = path.join(DEFAULTS.ARTIFACTS_FOLDER, configFile.file)
+      const { values } = loadConfigFile(configFilePath)
+
+      const runtimeManifest = getObjectValue(values, configFile.key)
+      for (const [, pkgManifest] of Object.entries(runtimeManifest.packages)) {
+        // key is the package name (unused), value is the package manifest. we iterate through each package's "actions"
+        for (const [, actionManifest] of Object.entries(pkgManifest.actions)) {
+          // key is the action name (unused), value is the action manifest. we add the "code-download: false" annotation
+          actionManifest.annotations['code-download'] = false
```

Comment:
nitpicking but should we keep true if the user explicitly sets it to true? (although I wouldn't see why)

Reply:
`aio app pack` as proposed is meant for ISV use, which means `code-download` annotation will always need to be false, which is a requirement. If we want to honor the existing annotation, that is possible, but not sure if that was the intended design. Will bring it up in Slack.

Reply:
Resolved based on internal Slack discussion: we always override with `false`. If a developer wants to make their ISV app-package downloadable on Exchange, then they can choose to do so in DD during the app submission.